variables.tf

```diff
@@ -0,0 +1,108 @@
+variable "ecr_repository_name" {
+  description = "Name of the ecr repository to hold lambda image"
+  type        = string
+  default     = "my-tf-function"
+}
+
+variable "security_group_id" {
+  description = "id of security group associated with ec2 deployment"
+  type        = string
+  default     = null
+}
+
+variable "public_subnet_ids" {
+  description = "ids of public subnet associated with ec2 deployment"
+  type        = list(string)
+  default     = []
+}
+
+variable "aws_region" {
+  description = "region to provision aws suite"
+  type        = string
+  default     = "us-east-1"
+}
```

Comment:
Instead of making this a variable, obtain it from the provider. Not doing this could result in a mismatch in the provider and the desired AWS region deployment.

Rough steps:

1. Remove this.
2. Get the [current region](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/region) (in `main.tf`).

    ```hcl
    data "aws_region" "current" {}
    ```

3. Replace all instances of `var.aws_region` with `data.aws_region.current`.